메인 페이지, 닉네임 설정, 프로필 사진 설정 페이지 UI

nimo-my opened the pull request:
main페이지 : UI 적용
main/welcome : 닉네임 설정
main/welcome2 : 프로필 사진 설정 페이지

등을 만들었습니다! 마지하고 싶어요~~ 엉엉

nimo-my committed:
로그인 후, 닉네임 입력창 @main/welcome 폴더안에 제작, 메인화면 새로 제작 #70

nimo-my committed:
welcome2 페이지 작성, 스크롤할때 스크롤 이벤트로 어떤 캐릭터 셀렉했는지 알려주는 함수 만들어야 함 #70

nimo-my committed:
chanhpar님이 추가해주신 이름나오는 코드#70

nimo-my committed:
UiFrame 을 UIFrame으로 수정 #70

nimo-my committed:
welcome2 페이지의 기능을 개선함 #70

React의 메모이제이션을 최대한 활용하도록 함.
IntersectionObserver의 핸들러에서 isIntersecting=false인 값을 무시하도록 함.

chanha-park commented on front/components/UIFrame/UiFrame.tsx:
저희 이거 파일이름은 그대로 UIFrame.tsx가 아닌 UiFrame.tsx로 하나요? UI가 줄임말인데 이런경우의 파스칼케이스는 어떻게 해야하죠.....

16EB replied:
저는 대문자 유지해서 사용하고 있습니다. Uuid 대신 UUID 같이요. 근거는 여기입니다.
https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID

chanha-park commented on front/app/main/@main/QuickMatch.tsx:
```diff
@@ -0,0 +1,43 @@
+            // href="/main/game/1234"
+            className="relative flex place-items-center"
+        >
+            <div className="gradient-border group relative flex w-[262px] flex-col items-start gap-2 rounded-[28px] border-transparent bg-windowGlass/30 p-px px-5 py-4 backdrop-blur-[50px] transition-colors before:pointer-events-none before:absolute before:inset-0 before:rounded-[28px] before:p-px before:backdrop-brightness-100 before:content-[''] hover:bg-primary focus:outline-2 focus:outline-gr ...
```
CreateGame과 QuickMatch 둘 공통된 이야기인데요, 
`<button><div></div></button>`처럼 기능 태그가 밖에, 서식을 위한 태그가 안에 있는 것과, 반대로 `<div><button></button></div>` 처럼 버튼이 안에 있는 것 중 어느 쪽이 낫나요? 저는 개인적으로는 기능부분이 안에 있고 스타일링 태그가 밖에 있는게 더 예뻐보이긴 하는데, 생각해보면 ref전달이나 로직 구현상 스타일링 태그가 안에 있는게 더 나을지도 모르겠다 싶네요.

chanha-park commented on front/app/main/@main/page.tsx:
```diff
@@ -1,30 +1,50 @@
+                </div>
+            </div>
+            <div className="flex flex-col">
+                <div className="pb-[46px] text-center text-[24px] font-bold not-italic leading-[45px] text-white">
```
위쪽 부분과 달리 기본 폰트가 italic이 아닌데 혹시 not-italic 태그 빼도 되나요..? 빼보니까 제가 보기엔 다른 것 없어보이긴 하는데... 명시적으로 써주는게 좋나요

chanha-park commented on front/app/main/@main/page.tsx:
```diff
@@ -1,30 +1,50 @@
+import { DoubleSharp } from "@/components/ImageLibrary";
+import { Game_Ghost3 } from "@/components/ImageLibrary";
+
+export function HelloWorld() {
```
이거 hello world 컴포넌트 밖에서 안쓰는거면 export 빼도 되지 않을까 싶습니다.
안쓰는 import들도요

16EB committed:
Update and rename CreateGame.tsx to CreateGameButton.tsx

chanha-park commented on front/app/main/@main/page.tsx:
```diff
@@ -1,30 +1,50 @@
+
+export function HelloWorld() {
+    return (
+        <div className="z-10 min-w-max">
```
투명도 설정이 뭔가 잘못되었는지 본 페이지에서 우측상단 채팅모달 버튼을 눌렀을 때 HelloWorld 컴포넌트 전체가 안가려지고 그대로 보이는 상황인데요. 음 이게 가리는 측이 아니고 가려지는쪽이 안가려지면 뭘 봐야되는건가요....?

16EB replied:
호두님이 고쳐주셨습니다. d60990b

chanha-park reviewed (commented):
몇가지 질문사항들 군데군데 댓글로 달았습니다,,

16EB committed:
전체적으로 한번 다듬었습니다.

UIFrame 대신 Card라는 이름을 사용하도록 했습니다.
UIFrame이 default export 대신 named export를 사용하게 했습니다.
Welcome2 페이지에서 사용한 의존성 배열을 linter가 알려주는대로 변경했습니다.
게임 버튼을 통합했습니다. 추후 컴포넌트로 만들 수 있어보입니다.

16EB committed:
호두's fix

16EB reviewed (approved):
모든 코드를 확인했습니다.

chanha-park reviewed (approved):
모두 확인했습니다 감사합니다
